Busted

macejmic opened the pull request.

Commit:
busted tests

macejmic commented on src/apicast-cli/cli/busted.lua:
```diff
@@ -0,0 +1,16 @@
+  local busted_run = parser:command("b busted", "Runs busted tests.")
+    :action(actions.busted.run)
+
+  busted_run:option("-p", "only run test files matching the Lua pattern")
```
@mikz @hallelujah 

don't know how to get this option in `src/apicast-cli/cmd/busted.lua` 

i'm doing `bin/apicast-cli b -p somepatter` but `args` are blank.

any idea?

mikz replied:
So this option is not needed anymore, right?

macejmic replied:
it is, to defined all available options.. otherwise it says:

```
bin/apicast-cli b -f ddddasdasd -r 10

Usage: apicast-cli b [-h]

Error: unknown option '-f'
Did you mean '-h'?
```

mikz replied:
I see. We really don't have to parse those options here. Maybe there is some setting to ignore unknown ones?

macejmic replied:
will check.

mikz commented on src/apicast-cli/cmd/busted.lua:
```diff
@@ -0,0 +1,19 @@
+local path = require "pl.path"
+
+local run_tests = function (args, _) 
+  if ngx ~= nil then
```
This is also not needed. Or the luarocks.loader.

macejmic replied:
ok

mikz committed:
remove unnecessary code

mikz committed:
forward all args to busted

mikz committed:
Merge remote-tracking branch 'origin/master' into busted

mikz committed:
generate customized specs with the project name

mikz committed:
Merge remote-tracking branch 'origin/master' into busted

mikz committed:
add missing files to the rockspec

mikz committed:
run busted_wrapper by resty command

because copied luafiles do not cary binary flags

so it could not be execed when packed as rock

macejmic commented:
🍤
